optuna/importance/_ped_anova/evaluator.py

```diff
@@ -0,0 +1,235 @@
+        target, min_n_top_trials = self._target, self._min_n_top_trials
+        sign = 1.0 if self._is_lower_better else -1.0
+        loss_values = sign * np.asarray([t.value if target is None else target(t) for t in trials])
+        err_msg = "len(trials) must be larger than or equal to min_n_top_trials"
+        assert min_n_top_trials <= loss_values.size, err_msg
+
+        def _quantile(v: np.ndarray, q: float) -> float:
+            cutoff_index = int(np.ceil(q * loss_values.size)) - 1
+            return float(np.partition(loss_values, cutoff_index)[cutoff_index])
+
+        cutoff_val = max(
+            np.partition(loss_values, min_n_top_trials - 1)[min_n_top_trials - 1],
+            # TODO(nabenabe0928): After dropping Python3.7, replace below with
+            # np.quantile(loss_values, self._quantile, method="inverted_cdf")
+            _quantile(loss_values, self._quantile),
+        )
+        should_keep_trials = loss_values <= cutoff_val
+        return [t for t, should_keep in zip(trials, should_keep_trials) if should_keep]
+
+
+@experimental_class("3.6.0")
+class PedAnovaImportanceEvaluator(BaseImportanceEvaluator):
+    """PED-ANOVA importance evaluator.
+
+    Implements the PED-ANOVA hyperparameter importance evaluation algorithm.
+
+    PED-ANOVA fits Parzen estimators of :class:`~optuna.trial.TrialState.COMPLETE` trials better
+    than a user-specified baseline. Users can specify the baseline by a quantile.
+    The importance can be interpreted as how important each hyperparameter is to get
+    the performance better than baseline.
+
+    For further information about PED-ANOVA algorithm, please refer to the following paper:
+
+    - `PED-ANOVA: Efficiently Quantifying Hyperparameter Importance in Arbitrary Subspaces
+      <https://arxiv.org/abs/2304.10255>`_
+
+    .. note::
+
+        The performance of PED-ANOVA depends on how many trials to consider above baseline.
+        To stabilize the analysis, it is preferable to include at least 5 trials above baseline.
+
+    .. note::
+
+        Please refer to `the original work <https://github.com/nabenabe0928/local-anova>`_.
+
+    Args:
+        baseline_quantile:
+            Compute the importance of achieving top-`baseline_quantile` quantile objective value.
+            For example, `baseline_quantile=0.1` means that the importances give the information
+            of which parameters were important to achieve the top-10% performance during
+            optimization.
+        evaluate_on_local:
+            Whether we measure the importance in the local or global space.
+            If :obj:`True`, the importances imply how importance each parameter is during
+            optimization. Meanwhile, `evaluate_on_local=False` gives the importances in the
+            specified search_space. `evaluate_on_local=True` is especially useful when users
+            modify search space during optimization.
+
+        Example:
```

Comment:
Could you please reduce the indentation by one level? In the generated document, the example code is included in the arguments section.

<img width="1050" alt="Screenshot 2024-02-19 at 16 58 06" src="https://github.com/optuna/optuna/assets/38826298/aa2ec43a-ec37-43f2-b005-e17613a82c3c">